checker/tests/nullness-ignoredeadcode/DeadBranchNPE.java

```diff
@@ -0,0 +1,31 @@
+class DeadBranchNPE {
+    void test1() {
+        Object obj = null;
+        if (true) {
+            // :: error: (dereference.of.nullable)
+            obj.toString();
+        } else {
+            // TODO: This is a dead branch should not issue error, the currently it does
+            // obj.toString();
+        }
+    }
+
+    void test2() {
+        Object obj = null;
+        // :: error: (dereference.of.nullable)
+        obj.toString();
+        // The following loop is dead code because the loop condition is false.
+        for (int i = 0; i < 0; i++) {
+            obj.toString();
```

Comment:
Also note that this also doesn't produce a warning before this PR - so does this PR do anything?
The deref on `obj` before is remembered and then we don't produce multiple errors. So, here and below, don't use the same variable outside the loop and inside.

Reply:
You should add the same test to the normal Nullness Checker tests and show the actual difference in behavior between the two.

Reply:
I see. I thought we want to enable the check first. I will write dead branch analysis in this PR. I think Jenny's thesis contains case study for dead branch analysis but not sure why it is not in the dataflow framework.